tests/testsuite/pub_priv.rs

```diff
@@ -479,3 +479,220 @@ fn allow_priv_in_custom_build() {
         )
         .run()
 }
+
+// A indirectly add D as private dependency.
```

Comment:
I've been unsure whether these should be merged before the related fix.  Thinking on it more, and with rust-lang/rust#119428 being moved to the rust repo, I suspect we should hold off on tests that demonstrate bad behavior.  The problem is that the fix would be merged into the rust repo but they can't update the tests in the same commit, adding more frustration to the process of fixing the bugs.

Sorry that I hadn't thought of it before you had put more time into polishing this up.

Reply:
It is ok to close this PR or add WIP to the title until the issue fixed in rust repo.